### PR TITLE
Add set_slot_new_foreign_scratch to allow users to specify a cratch slot when creating a foreign object

### PR DESCRIPTION
**Pull request by pac85**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1280,6 +1280,73 @@ impl VM {
...
 
+    pub fn set_slot_new_foreign_scratch<M: AsRef<str>, C: AsRef<str>, T: 'static + ClassObject>(
+        &self,
+        module: M,
+        class: C,
+        object: T,
+        slot: SlotId,
+        scratch: SlotId,
+    ) -> Result<&mut T, ForeignSendError> {
+        self.ensure_slots(slot + 1);
```
Comment (Jengamon) [inline src/lib.rs:1291]: Should this be `self.ensure_slots(scratch + 1)`?

File: src/lib.rs [hunk 2]
```diff
@@ -1280,6 +1280,73 @@ impl VM {
...
+                    // The Wren foreign class corresponds with this real object.
+                    // We can coerce it and treat this object as that class, even if not instantiated by Wren.
+
+                    // Create the new ForeignObject
+                    let new_obj = ForeignObject {
+                        object: Box::into_raw(Box::new(object)),
+                        type_id: any::TypeId::of::<T>(),
+                    };
+
+                    // Load the Wren class object into slot 0.
```
Comment (Jengamon) [inline src/lib.rs:1313]: For consistency, change this from slot 0 to scratch slot.

File: src/lib.rs [hunk 3]
```diff
@@ -1280,6 +1280,73 @@ impl VM {
...
+        module: M,
+        class: C,
+        object: T,
+        slot: SlotId,
+        scratch: SlotId,
+    ) -> Result<&mut T, ForeignSendError> {
+        self.ensure_slots(slot + 1);
+        let conf = unsafe { &mut *(wren_sys::wrenGetUserData(self.vm) as *mut UserData) };
+
+        self.ensure_slots((slot + 1) as usize);
```
Comment (Jengamon) [inline src/lib.rs:1294]: I'm not sure if we need the `usize` cast here. (I might have just written it that way, and if I did, you can just resolve this comment.)

**Review by Jengamon — CHANGES_REQUESTED** [review]
Looking really good! Just a few small adjustments.

File: src/lib.rs (hunk 3 above)
Comment (pac85) [reply src/lib.rs]: Yes it was already there

**Review by Jengamon — APPROVED** [review]

**Comment by Jengamon** [comment]
Gotta switch CI servers...

**Pull request merged**